apps/designer/app/canvas/canvas.tsx

```diff
@@ -123,7 +123,10 @@ const useCopyPaste = () => {
       publish({ type: "deleteInstance", payload: { id: instance.id } });
     },
     onPaste: (instance, props) => {
-      publish({ type: "insertInstance", payload: { instance, props } });
+      publish({
+        type: "insertInstance",
+        payload: { instance, props, isPopulated: true },
```

Comment:
I would split events. When drop from components list need to pass only component name so instance could be created in subscription.

Reply:
currrent way of doing it gives you the benefit of configuring the instance where its being created, might be better long-term

Reply:
actually populate instance shouldn't be even called inside insertInstance handler, it should be called where a new instance was created and needs to be populated